framework/include/GameBoard.h

```diff
@@ -0,0 +1,49 @@
+#ifndef GAMEBOARD_H
+#define GAMEBOARD_H
+
+#include "game.h"
+#include <string>
+
+namespace Game {
+
+// GameBoard class
+// Used to store and interface with the board state
+// Implementation is defined by the user depending on the game
+class GameBoard {
+public:
+
+    // Note: The following functions are constant for each game, do not modify
+    GameBoard() = default;
```

Comment:
Might be a little nitpicky, but the constructor and destructor here might could be protected to prevent someone from directly instantiating this object outside the derivation chain.

Reply:
Actually now that I've looked at it more, I don't think you want these to be protected.

Reply:
Yeah, this class isn't inherited, so making it protected means it can't be instantiated at all.